apps/sms/test/marionette/notification_test.js

```diff
@@ -225,7 +226,63 @@ marionette('Message notification tests', function() {
 
       openNotification();
 
-      assertMessagesIsInCorrectState();
+      assertMessagesIsInCorrectStateFromMessage(smsMessage);
+    });
+  });
+
+  suite('Display a conversation from a notification', function() {
+    var messagesStorage;
+    var thread1, thread2;
+
+    setup(function() {
+      thread1 = ThreadGenerator.generate({
+        numberOfMessages: 50,
+        participants: ['999']
+      });
+      thread2 = ThreadGenerator.generate({
+        numberOfMessages: 50,
+        participants: ['888']
+      });
+
+      messagesStorage = [ thread1, thread2 ];
+    });
+
+    test('Clicking a notification while a conversation is loading', function() {
+      messagesApp.launch();
+      storage.setMessagesStorage(messagesStorage);
+      // Set empty contacts store.
+      storage.setContactsStorage();
+
+      var inbox = new InboxView(client);
+      var conversation = inbox.goToConversation(
+        thread1.id, { shouldWait: false }
```

Comment:
__question:__ mmmmm, I believe there is a chance we won't have the issue on such _small_ conversations in real life as well :) What about 200-300 messages in the first thread instead? Closer to real use case and I believe it should fail :)